### PR TITLE
Adding avg and mps speed formulae for ideal gases

### PR DESCRIPTION
**Pull request by Baron105**
### Describe your change:

Added functions to calculate the average and most probable speeds of molecules in a gas given the temperature and molar mass of the gas. Also added a few doctests for testing.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

**Comment by Baron105** [comment]
Incorporated all the suggestions in the file. Please check @tianyizheng02. Thank you for the detailed review.

File: physics/speeds_of_gas_molecules.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+More information about the Maxwell-Boltzmann distribution can be found here:
+https://en.wikipedia.org/wiki/Maxwell%E2%80%93Boltzmann_distribution
+
+The average speed can be calculated by integrating the Maxwell-Boltzmann distribution
+from 0 to infinity and dividing by the total number of molecules. The result is:
+
+        ---------------------
+        | vavg = √8RT/πM    |
+        ---------------------
```
Comment (tianyizheng02) [inline physics/speeds_of_gas_molecules.py:26]:
```suggestion
        ---------------------
        | vavg = √(8RT/πM)  |
        ---------------------
```
For clarity

File: physics/speeds_of_gas_molecules.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+The most probable speed is the speed at which the Maxwell-Boltzmann distribution
+is at its maximum. This can be found by differentiating the Maxwell-Boltzmann
+distribution with respect to v and setting the result equal to zero. The result is:
+
+        ---------------------
+        | vmp = √2RT/M      |
+        ---------------------
```
Comment (tianyizheng02) [inline physics/speeds_of_gas_molecules.py:34]:
```suggestion
        ---------------------
        | vmp = √(2RT/M)    |
        ---------------------
```

File: physics/speeds_of_gas_molecules.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+The root-mean-square speed is another measure of the average speed
+of the molecules in a gas. It is calculated by taking the square root
+of the average of the squares of the speeds of the molecules. The result is:
+
+        ---------------------
+        | vrms = √3RT/m     |
+        ---------------------
```
Comment (tianyizheng02) [inline physics/speeds_of_gas_molecules.py:42]:
```suggestion
        ---------------------
        | vrms = √(3RT/M)   |
        ---------------------
```

File: physics/speeds_of_gas_molecules.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+The most probable speed is the speed at which the Maxwell-Boltzmann distribution
+is at its maximum. This can be found by differentiating the Maxwell-Boltzmann
+distribution with respect to v and setting the result equal to zero. The result is:
+
+        ---------------------
+        | vmp = √2RT/M      |
+        ---------------------
+
+The root-mean-square speed is another measure of the average speed
```
Comment (tianyizheng02) [inline physics/speeds_of_gas_molecules.py:36]: No function for the RMS speed?

File: physics/speeds_of_gas_molecules.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
+"""
+The root-mean-square, average and most probable speeds are derived from
+the Maxwell-Boltzmann distribution. The Maxwell-Boltzmann distribution is a
+probability distribution that describes the distribution of speeds for particles
+in a gas.
```
Comment (tianyizheng02) [inline physics/speeds_of_gas_molecules.py:5]:
```suggestion
The root-mean-square, average and most probable speeds of gas molecules
are derived from the Maxwell-Boltzmann distribution. The Maxwell-Boltzmann
distribution is a probability distribution that describes the distribution of
speeds of particles in an ideal gas.
```
A few minor additions for clarity

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

**Comment by Baron105** [comment]
@tianyizheng02 please check if the current file is ok

**Review by tianyizheng02 — APPROVED** [review]

**Comment by tianyizheng02** [comment]
> I did not add the rms function as there is already a file that defines the function ie physics/rms_speed_of_molecule.py as that would be repeating the function. Should I add it here for a complete comparison of all three speeds?

No need, I forgot that we already had a file for the RMS speed. However, it'd make more sense in my opinion to move the existing RMS speed code into your code file since they're all about gas molecule speeds—that could be work for a later PR.

**Comment by tianyizheng02** [comment]
Thanks for your contribution!

**Comment by Baron105** [comment]
> Thanks for your contribution!

Glad to contribute! Thank you for taking time to review the file and giving great advice!

**Pull request merged**